### ISSUE TITLE
Fatal error when an HTML comment is inside an HTML file in 3.0.0

### ISSUE THREAD

**Issue by renatoi** [ember-template-lint/ember-template-lint#1718]
The following code causes the fatal error:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>title</title>
  </head>
  <body>
    <!-- FOO -->
    <p>Hello World</p>
  </body>
</html>
```

Error:

```
  1:0  error  Closing tag `body` (on line 10) without an open tag.  undefined

✖ 1 problems (1 errors, 0 warnings)
error Command failed with exit code 1.
```

Removing `<!-- FOO -->` fixes it.

This shouldn't happen.

**Comment by rwjblue** [comment]
It's due to DOCTYPE I believe. We need to strip it before parsing (then bring it back before writing when fixing).

**Comment by rwjblue** [comment]
https://github.com/tildeio/simple-html-tokenizer/pull/71 has related info (where I was trying to add built in parser support for doctype).

**Comment by locks** [comment]
I just tried to reproduce in the current version and confirmed it is fixed!
